py-polars/polars/dataframe/frame.py

```diff
@@ -4381,28 +4381,39 @@ def filter(
             Each constraint will behave the same as `pl.col(name).eq(value)`, and
             will be implicitly joined with the other filter conditions using `&`.
 
+        Notes
+        -----
+        If you are transitioning from pandas and performing filter operations based on
+        the comparison of two or more columns, please note that in Polars,
+        any comparison involving null values will always result in null.
+        As a result, these rows will be filtered out.
+        Ensure to handle null values appropriately to avoid unintended filtering
```

Comment:
I think you mention `ne_missing` and `eq_missing`, and I'd rather see an example using those methods as those will include missing values in the comparison.